### ISSUE TITLE
Update Cupertino Theme to match icon sizes to MediaQuery.textScaleFactor

### ISSUE THREAD

**Issue by DanielEdrisian** [flutter/flutter#67792]
SF symbol sizes match the system font scales. Changing font scales through accessibility settings should also change the icon sizes. Could modify the cupertino theme to handle this:

https://github.com/flutter/flutter/blob/439992472a61f2062f106f8a421aead288d7038c/packages/flutter/lib/src/cupertino/theme.dart#L102

**Comment by xster** [comment]
We should also check whether this behavior is present in Android and we just omitted it or whether we want to make this platform specific. 

Perhaps @goderbauer already knows the answer for Android.

**Comment by goderbauer** [comment]
I don't know how Android behaves here. We need to check.

**Comment by xster** [comment]
So it's a pretty complicated answer. I took a look at Android and Android's font scale API (https://developer.android.com/reference/android/content/res/Configuration#fontScale) definitely doesn't affect icon sizes at all. 

Android actually has an explicit "display size" accessibility setting which full-on changes the DPI of the device and thus changes everything from text to icon to the rest of the UI. 

Just as I was having a headache thinking about how to map it to iOS, I took a quick look and it turns out iOS doesn't really support dynamic icon sizes based on accessibility font scales either. 

Here are some examples:
![Screen Shot 2020-10-13 at 2 29 55 PM](https://user-images.githubusercontent.com/156888/95918392-91426080-0d60-11eb-8b2e-76d6f93db010.png)

Here, almost all the icons are static in size except the search bar in maps and the circled plus button in imessage. In other words, what's happening here is less about icons matching accessibility setting but more of an inconsistent application of intrinsic size vs size by parent in various places with icons (especially see in settings the leading vs trailing icons). 

So here, I think the general rule is that the icon size is independent, and we have exceptions to the rule such as in the search bar where the font determines the size of the container which determines the size of the icon. 

Let's just treat that as a one off and read the text scale in CupertinoSearchBar specifically (strictly speaking we would be more correct in having a LayoutBuilder build the text in the middle, figure out its desired intrinsic size, then come back and build the icons around it with icon sizes based on that text field intrinsic size but we're just pointlessly penalizing our performance since that text field size is just based on text scale to start with anyway).
